cl/opinion_page/views.py

```diff
@@ -948,6 +1098,159 @@ async def view_authorities(
+        doc_type="opinion",
+    )
+    await authorities_context.post_init()
+
+    additional_context = {
+        "authorities_context": authorities_context,
+        "authorities_with_data": await cluster.aauthorities_with_data(),
+    }
+    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
+        request, "ui_flag_for_o"
+    )
+    user_flag_active = await sync_to_async(waffle.flag_is_active)(
+        request.user, "ui_flag_for_o"
+    )
+
+    if ui_flag_for_o or user_flag_active:
+        return await render_opinion_view(
+            request, pk, "authorities", additional_context
+        )
+    else:
+        # Old page to load for people outside the flag
+        return await view_authorities(
+            request=request, pk=pk, slug="authorities"
+        )
+
+
+async def view_opinion_cited_by(
+    request: HttpRequest, pk: int, _: str
+) -> HttpResponse:
+    """"""
+    cluster: OpinionCluster = await aget_object_or_404(OpinionCluster, pk=pk)
+
+    (
+        citing_clusters,
+        citing_cluster_count,
+        _,
+    ) = await es_get_cited_clusters_with_cache(cluster, request)
+    additional_context = {
+        "citing_clusters": citing_clusters,
+        "citing_cluster_count": citing_cluster_count,
+    }
+    return await render_opinion_view(
+        request, pk, "cited-by", additional_context
+    )
+
+
+async def view_opinion_summaries(
+    request: HttpRequest, pk: int, _: str
+) -> HttpResponse:
+    """"""
+    cluster: OpinionCluster = await aget_object_or_404(OpinionCluster, pk=pk)
+    parenthetical_groups_qs = await get_or_create_parenthetical_groups(cluster)
+    parenthetical_groups = [
+        parenthetical_group
+        async for parenthetical_group in parenthetical_groups_qs.prefetch_related(
+            Prefetch(
+                "parentheticals",
+                queryset=Parenthetical.objects.order_by("-score"),
+            ),
+            "parentheticals__describing_opinion__cluster__citations",
+            "parentheticals__describing_opinion__cluster__docket__court",
+            "representative__describing_opinion__cluster__citations",
+            "representative__describing_opinion__cluster__docket__court",
+        )
+    ]
+    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
+        request, "ui_flag_for_o"
+    )
+    user_flag_active = await sync_to_async(waffle.flag_is_active)(
+        request.user, "ui_flag_for_o"
+    )
+
+    if ui_flag_for_o or user_flag_active:
+        additional_context = {
+            "parenthetical_groups": parenthetical_groups,
+            "ui_flag_for_o": ui_flag_for_o,
+            "user_flag_active": user_flag_active,
+        }
+        return await render_opinion_view(
+            request, pk, "summaries", additional_context
+        )
+    else:
+        # Old page to load for people outside the flag
+        return await view_summaries(request=request, pk=pk, slug="summaries")
```

Comment:
Similar suggestion for the Waffle flag.

```suggestion
    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
        request, "ui_flag_for_o"
    )
    if not ui_flag_for_o:
        # Old page to load for people outside the flag
        return await view_summaries(request=request, pk=pk, slug="summaries")

    additional_context = {
        "parenthetical_groups": parenthetical_groups,
        "ui_flag_for_o": ui_flag_for_o,
    }
    return await render_opinion_view(
        request, pk, "summaries", additional_context
    )
```